components/concourse-task-toolbox/bin/determine-platform-version.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+import sys
+
+os.makedirs("platform-version", exist_ok=True)
+
+print("Picking platform version...")
+partial_repos = [
+    "platform",
+    "service-operator-source",
+    "concourse-task-toolbox-source",
+    "concourse-operator-source",
+    "concourse-github-resource-source",
+    "concourse-harbor-resource-source"
+]
+commits_known_by = {open(f"{r}/.git/ref").read(): [] for r in partial_repos}
```

Comment:
I think what we are aiming for, in pseudocode, is:
```
repo_map = {}
foreach partial_repo:
    if partial_repo.branch != master:
        exit non-zero
    # if multiple repos reference the same commit, the commit_count will also match
    repo_map[partial_repo.commit] = partial_repo.commit_count
repo_map.order_by(commit_count, DESC)
commit_we_care_about = repo_map.first
```

Reply:
Oh I see, so you suggest we just take the one with the highest number of commits in its history for master?

Reply:
Yes, assuming "the most recent" is the one we care about...